dashboard/app/models/script_level.rb

```diff
@@ -556,4 +560,39 @@ def hidden_for_section?(section_id)
   def should_hide_survey(user, viewed_user)
     anonymous? && user.try(:teacher?) && !viewed_user.nil? && user != viewed_user
   end
+
+  def seeding_key(seed_context, use_existing_level_keys = true)
+    my_key = {
+      'script_level.level_keys': get_level_keys(seed_context, use_existing_level_keys),
+      'script_level.chapter': chapter,
+      'script_level.position': position
+    }
+
+    my_lesson = seed_context.lessons.select {|l| l.id == stage_id}.first
+    raise "No Lesson found for #{self.class}: #{my_key}, Lesson ID: #{stage_id}" unless my_lesson
+    lesson_seeding_key = my_lesson.seeding_key(seed_context)
+
+    my_key.merge!(lesson_seeding_key) {|key, _, _| raise "Duplicate key when generating seeding_key: #{key}"}
+    my_key.stringify_keys
+  end
+
+  # TODO: add validation so this is always in sync?
+  def get_level_keys(seed_context, use_existing_level_keys = true)
+    return self.level_keys if use_existing_level_keys && !self.level_keys.nil_or_empty? # rubocop:disable Style/RedundantSelf
+
+    if levels.loaded?
```

Comment:
what does `loaded?` do? I'm not sure I am looking at the right docs: https://apidock.com/rails/ActiveRecord/Associations/Association/loaded%3F

Reply:
I think that's right. It returns true if the associated object(s) are already in memory.